src/js/components/CheckBox/CheckBox.js

```diff
@@ -30,50 +34,105 @@ class CheckBox extends Component {
       ...rest
     } = this.props;
 
-    const normalizedLabel = (typeof label === 'string' ? <div>{label}</div> : label);
-
     let hidden;
     if (disabled && checked) {
       hidden = <input name={name} type='hidden' value='true' />;
     }
 
     const Icon = theme.checkBox.icons.checked;
 
+    let borderColor = normalizeColor(
+      theme.checkBox.border.color[theme.dark ? 'dark' : 'light'],
+      theme
+    );
+    if (checked) {
+      borderColor = normalizeColor(
+        (
+          theme.checkBox.color || theme.global.control.color
+        )[theme.dark ? 'dark' : 'light'],
+        theme
+      );
+    }
+
     const visual = (toggle ? (
-      <StyledCheckBoxToggle focus={focus} theme={theme}>
+      <StyledCheckBoxToggle focus={focus} theme={theme} checked={checked}>
         <StyledCheckBoxKnob theme={theme} />
       </StyledCheckBoxToggle>
     ) : (
-      <StyledCheckBoxBox focus={focus} theme={theme}>
+      <StyledCheckBoxBox
+        as={Box}
+        align='center'
+        justify='center'
+        width={theme.checkBox.size}
+        height={theme.checkBox.size}
+        border={{
+          size: theme.checkBox.border.width,
+          color: evalStyle(borderColor, theme),
+        }}
+        round={theme.checkBox.check.radius}
```

Comment:
Can we relabel to `theme.checkBox.round`? I don't think the `check` has a `radius`.

Reply:
this is not the check icon, this is the box surrounding the icon. I think for the same reason as above, keep radius. we are using radius for the other controls.